src/v/cluster/ntp_callbacks.h

```diff
@@ -153,8 +153,8 @@ class ntp_callbacks {
 
     template<typename... Args>
     void notify(const callbacks_t& callbacks, Args&&... args) const {
-        for (auto& cb : callbacks) {
-            cb.second(std::forward<Args>(args)...);
+        for (auto cb_i = callbacks.cbegin(); cb_i != callbacks.cend();) {
+            (cb_i++)->second(std::forward<Args>(args)...);
```

Comment:
how does this not fall into the category of erasure-invalidates-iterator?

Reply:
It's a bit finicky. I don't believe it falls into the category though. The behavior for `absl::flat_hash_map` differs from std maps in that the only the iterator to the element that is erased is invalidated; at least that's the documented behavior [here](https://github.com/abseil/abseil-cpp/blob/master/absl/container/flat_hash_map.h#L228-L233). And the C++ post increment operator will increment the iterator first then return a copy of the iterator pre-incrementing. So the iterator is always incremented before the element it is pointed to can be erased and erasures only invalidate iterators that point to the erased element.